src/Controller/HomeController.php

```diff
@@ -3,17 +3,52 @@
 namespace App\Controller;
 
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class HomeController extends AbstractController
 {
     /**
-     * @Route("/", name="home")
+     * @Route("/", name="student_list")
+     * @param KernelInterface $kernel
+     * @return Response
      */
-    public function index()
+    public function index(KernelInterface $kernel)
     {
+        $json = (string)file_get_contents($kernel->getProjectDir().'/public/students.json');
```

Comment:
```suggestion
        $json = file_get_contents($kernel->getProjectDir().'/public/students.json');
        
        if ($json === false) {
           throw \Exception("Unable to read internal file");
        }
```

PHP pasaulyje skirtingi tipai grąžinami, kad būtų galima reaguoti į klaidas.

Konvertuodamas į kitą tipą ne labai ką išloši:
```php
$input = (string) false;
var_dump($input); 
var_dump(json_decode($input, true));
````
> string(0) ""
> NULL
```php
$input = (string) false;
var_dump($input); 
var_dump(json_decode($input, true));
```
> bool(false)
> NULL
T.y. vis tiek gausi klaidą:
> PHP Warning:  Invalid argument supplied for foreach()

Todėl geriau arba lūžti (su Exception) iš karto, arba užtikrinti, kad sistema gali pasigyditi pati (`if ($json === false) { $json = '{}' }`)

Plačiau:
 * https://www.php.net/manual/en/function.file-get-contents.php
 * https://en.wikipedia.org/wiki/Fail-fast